packages/components/src/DataList/components/DataListHeaderTile/DataListHeaderTile.tsx

```diff
@@ -0,0 +1,56 @@
+import React from "react";
+import classnames from "classnames";
+import styles from "./DataListHeaderTile.css";
+import { SortingArrows } from "./SortingArrows";
+import { Text } from "../../../Text";
+import { useDataListContext } from "../../context/DataListContext";
+import { DataListHeader, DataListObject } from "../../DataList.types";
+
+interface DataListHeaderTileProps<T extends DataListObject> {
+  headers: DataListHeader<T>;
+  headerKey: string;
+}
+
+export function DataListHeaderTile<T extends DataListObject>({
+  headers,
+  headerKey,
+}: DataListHeaderTileProps<T>) {
+  const { sorting } = useDataListContext();
+
+  const isSortable = sorting?.sortable.includes(headerKey);
+  const sortingState = sorting?.state;
+
+  function toggleSorting(sortingKey: string) {
+    if (sortingState?.direction === "desc") {
+      sorting?.onSort(undefined);
+      return;
+    }
+
+    sorting?.onSort({
+      direction: sortingState?.direction === "asc" ? "desc" : "asc",
+      key: sortingKey,
+    });
+  }
+
+  function handleOnClick() {
+    if (!isSortable) return;
+
+    toggleSorting(headerKey);
+  }
```

Comment:
Consistency-wise, drop these boys down below return. Pretty minor thing but reading the JSX first makes it easier to understand what this things are for